crates/edr_provider/src/snapshot.rs

```diff
@@ -0,0 +1,20 @@
+use std::time::Instant;
+
+use edr_eth::{Address, B256, U256};
+use edr_evm::{
+    state::{IrregularState, StateError, SyncState},
+    MemPool,
+};
+
+pub struct Snapshot {
+    pub block_number: u64,
```

Comment:
What's the block number used for? Is there anything else needed from the blockchain?

Reply:
This was caught using your destructing suggestion too. Resolved in https://github.com/NomicFoundation/hardhat/pull/4617/commits/46667e4b42d5973a03db6151f43b48a97a1a83f6 Thanks!